formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/services/FormstackService.scala

```diff
@@ -125,22 +127,22 @@ import FormstackService._
       }
 
       decodeIfNotSkippableError[Submission](response).map{
-        case None => Skipped(submissionIdEmail)
+        case None => Left(submissionIdEmail)
         case Some(submission) =>
           //validate the submission we found in formstack actually contains references to the email we were looking for
-          if (validateEmail(submissionIdEmail.email, submission))
-            Found(submission)
+          if (validateEmail(requestEmail, submission))
+            Right(submission)
```

Comment:
the bug I found was that this `submissionIdEmail.email` is actually a one way hash of the email as stored in the db, not the actual email.
The solution I came up with when I found this bug late yesterday was to pass around the original request email along with all the submission ids we got from the db and use that to validate that everything we get from formstack contains that email. Another option would have been to pass around the dependencies we need to do the hash and hash what we get from formstack and compare the two hashes instead